front/src/applications/stdcm/components/StdcmResults/SendToRailwayManagerModal.tsx

```diff
@@ -0,0 +1,430 @@
+import { Download, CheckCircle } from '@osrd-project/ui-icons';
+import { PDFDownloadLink, pdf } from '@react-pdf/renderer';
+import { useTranslation } from 'react-i18next';
+
+import {
+  type StdcmSimulationInputs,
+  type StdcmSuccessResponse,
+  type SimilarTrainWithSecondaryCode,
+  type StdcmResultsOperationalPoint,
+} from 'applications/stdcm/types';
+import {
+  osrdRailwayManagerApi,
+  type BodyLastMinuteRequest,
+} from 'common/api/osrdRailwayManagerApi';
+import MAIN_API from 'config/config';
+import {
+  createLinkedTrainPayload,
+  createSimilarTrainPayload,
+  getStopDurationTime,
+  transformStepsToApiFormat,
+  validateTolerance,
+} from 'modules/SimulationReportSheet/utils/formatSimulationReportSheet';
+import type { StdcmPathStep } from 'reducers/osrdconf/types';
+import { useDateTimeLocale } from 'utils/date';
+import type { DeploymentSettings } from 'utils/hooks/useDeploymentSettings';
+import { kmhToMs, minToMs, msToMin, tToKg } from 'utils/physics';
+
+import StdcmSimulationReportSheet from './StdcmSimulationReportSheet';
+
+type SendToRailwayManagerModalProps = {
+  onClose: () => void;
+  consist: StdcmSimulationInputs['consist'];
+  stdcmData: StdcmSuccessResponse;
+  linkedTrains: StdcmSimulationInputs['linkedTrains'];
+  simulationReportSheetNumber: string;
+  operationalPointsList: StdcmResultsOperationalPoint[];
+  simulationSheetLogo?: string;
+  similarTrains: SimilarTrainWithSecondaryCode[];
+  deploymentSettings?: DeploymentSettings;
+};
+
+type Option = { value: string; label: string };
+
+const SendToRailwayManagerModal = ({
+  consist,
+  onClose,
+  stdcmData,
+  linkedTrains,
+  simulationReportSheetNumber,
+  operationalPointsList,
+  simulationSheetLogo,
+  similarTrains,
+  deploymentSettings,
+}: SendToRailwayManagerModalProps) => {
+  const { t } = useTranslation('stdcm', { keyPrefix: 'simulation.results' });
+  const dateTimeLocale = useDateTimeLocale();
+
+  const steps = stdcmData?.simulationPathSteps ?? [];
+
+  const firstStep = steps[0] as Extract<StdcmPathStep, { isVia: false }>;
+  const lastStep = steps[steps.length - 1] as Extract<StdcmPathStep, { isVia: false }>;
+
+  const scheduledDepartureTime = new Date(stdcmData.departure_time);
+  const departureTime = new Date(firstStep.arrival!);
+
+  const intermediatePoints = useMemo(() => steps.slice(1, -1), [steps]);
+
+  const [pathType, setPathType] = useState<Option>();
+  const [comment, setComment] = useState('');
+  const [csCode, setCsCode] = useState('');
+  const diffMinutes = msToMin(scheduledDepartureTime.getTime() - departureTime.getTime());
+
+  const baseToleranceBefore = firstStep.tolerances.before.total('minute');
+  const baseToleranceAfter = firstStep.tolerances.after.total('minute');
+
+  const [beforeTolerance, setBeforeTolerance] = useState(
+    (baseToleranceBefore + diffMinutes).toString()
+  );
+  const [afterTolerance, setAfterTolerance] = useState(
+    (baseToleranceAfter - diffMinutes).toString()
+  );
```

Comment:
- `baseToleranceBefore` and `baseToleranceAfter` are not used in other places in the file, so I think you can remove the intermediate consts
- it would be better to handle numbers instead of strings
- you can remove all the cast to `Number` (`Number(...)`)

```suggestion
  const [beforeTolerance, setBeforeTolerance] = useState(
    firstStep.tolerances.before.total('minute') + diffMinutes
  );
  const [afterTolerance, setAfterTolerance] = useState(
   firstStep.tolerances.after.total('minute') - diffMinutes
  );
```